experimental/trusted_shuffler/trusted_shuffler/src/lib.rs

```diff
@@ -193,11 +224,31 @@ impl TrustedShuffler {
                      response_sender,
                  }| {
                     response_sender
-                        .send(data)
+                        .send(get_response(data))
                         .map_err(|_data| anyhow!("Couldn't send response"))
                 },
             )
             .collect::<Result<Vec<_>, _>>()
             .map(|_: Vec<()>| ())
     }
 }
+
+fn get_request(w: Wrapper) -> TrustedShufflerRequest {
+    match w {
+        Wrapper::Request(r) => r,
+        _ => panic!(
```

Comment:
I think panics are generally discouraged because they stop the whole program. It may be better to return an error.
Or in this case implement them as compile time checks (via generics and etc.)

Reply:
But it would probably be a separate PR